packages/backend-modules/access/lib/grants.js

```diff
@@ -433,54 +433,10 @@ const invalidate = async (grant, reason, t, pgdb, mail) => {
 
   await eventsLib.log(grant, `invalidated.${reason}`, pgdb)
 
-  if (grant.recipientUserId) {
-    const recipient = await pgdb.public.users.findOne({
-      id: grant.recipientUserId,
-    })
-
-    if (recipient) {
-      const campaign = await pgdb.public.accessCampaigns.findOne({
-        id: grant.accessCampaignId,
-      })
-
-      const hasRoleChanged = await membershipsLib.removeMemberRole(
-        grant,
-        recipient,
-        findByRecipient,
-        pgdb,
-      )
+  const sendMail = reason !== 'admin'
```

Comment:
This strikes me as too implicit, using a random, undocumented string.

Maybe check if `grant.recipientUserId !== context.user.id` (not recipient itself requested that) and `!context.user` (scheduler is invalidating)?